src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1707,6 +1793,40 @@ static async Task ExecuteAwaited(ValueTask task)
         return ExecuteAwaited(task);
     }
 
+    private static ValueTask<object?> ExecuteTaskWithEmptyResult(Task task)
+    {
+        static async ValueTask<object?> ExecuteAwaited(Task task)
+        {
+            await task;
+            return EmptyHttpResult.Instance;
+        }
+
+        if (task.IsCompletedSuccessfully)
+        {
+            task.GetAwaiter().GetResult();
```

Comment:
I'm pretty sure this isn't necessary with `Task`.

We always observe `ValueTask` in this situation to report back to `IValueTaskSource` that the caller is finished with the value task. `Task` doesn't have the same extensibility and a successfully completed task is fire and forget.